src/mono/mono.proj

```diff
@@ -339,8 +339,14 @@ JS_ENGINES = [NODE_JS]
           WorkingDirectory="$(ArtifactsObjDir)"
           IgnoreStandardErrorWarningFormat="true" />
 
-    <Copy SourceFiles="$(MSBuildThisFileDirectory)/wasi/wasi-sdk/share/wasi-sysroot/include/wasm32-wasi-threads/pthread.h"
-          DestinationFiles="$(MSBuildThisFileDirectory)/wasi/wasi-sdk/share/wasi-sysroot/include/wasm32-wasip2/pthread.h"
+    <!-- Temporary in WASI-SDK 22
+    TODO https://github.com/dotnet/runtime/issues/104773
+    -->
+    <Copy SourceFiles="$(WASI_SDK_PATH)/share/wasi-sysroot/include/wasm32-wasi-threads/pthread.h"
```

Comment:
If `WASI_SDK_PATH` is set by the CI agent to some read-only location, this will fail together with other patches we do here. I need to re-consider.

Reply:
Actually the provisioning and patching will be skipped.

Reply:
solved by adding `$(WASI_SDK_PATH)/VERSION22PATCHED` stamp file and checking it everywhere. Also to decide if we need to do provisioning